src/userService.ts

```diff
@@ -0,0 +1,134 @@
+import { users } from "./mockedData";
+import { Hobbies, User } from "./types";
+
+export class UserService {
+    private static instance: UserService | null = null;
+
+    private usersData: User[] = [];
+
+    private constructor() {
+        this.usersData = users;
+    }
+
+    public static getInstance(): UserService {
+        if (!UserService.instance) {
+            UserService.instance = new UserService();
+        }
+        return UserService.instance;
+    }
+
+    getAllUsers(): User[] {
+        return this.usersData;
+    };
+
+    getUserById(id: number): Partial<Omit<User, 'hobbies'>> | null {
+        if (id) {
+            const user = this.usersData.find(user => user.id === id) as Partial<User> || null;
+            const reternedUser = {...user};
+            if (user) {
+                delete reternedUser.hobbies
+            }
+            return reternedUser
+        }
+        return null
+        
+    };
+
+    getUserHobbies(id: number): Hobbies | null {
+        if (id) {
+            const user = this.usersData.find(user => user.id === id);
+            
+            return user?.hobbies || null
+        }
+        
+        return null
+    };
```

Comment:
I would suggest to rewrite it like this

```suggestion
    getUserHobbies(id: number): Hobbies | null {
        const user = this.usersData.find(user => user.id === id);
        return user?.hobbies || null
    };
```